### PR TITLE
Pretty float

### PR DESCRIPTION
**Pull request by kroening**
This PR makes the output of floating-point numbers look more similar to what stdc libraries do

### CODE REVIEW

File: src/util/ieee_float.cpp [hunk 1]
```diff
@@ -167,19 +167,25 @@ std::string ieee_floatt::format(const format_spect &format_spec) const
       mp_integer _exponent, _fraction;
       extract_base10(_fraction, _exponent);
 
-      if(_exponent>=0)
+      mp_integer adjusted_exponent=base10_digits(_fraction)+_exponent;
+
+      if(adjusted_exponent>=format_spec.precision ||
+         adjusted_exponent<-4)
```
Comment (tautschnig) [inline src/util/ieee_float.cpp]: Would it be possible to have a comment explaining the magic number `-4`?

File: src/util/ieee_float.cpp [hunk 2]
```diff
@@ -167,19 +167,25 @@ std::string ieee_floatt::format(const format_spect &format_spec) const
...
-          result+=to_string_scientific(format_spec.precision);
-        else
-          result+=to_string_decimal(format_spec.precision);
+        result+=to_string_decimal(format_spec.precision);
+
+        // Implementations tested also appear to suppress trailing zeros
+        // and trailing dots.
+
+        while(!result.empty() && result.back()=='0')
+          result.resize(result.size()-1);
```
Comment (tautschnig) [inline src/util/ieee_float.cpp]: I think I would have implemented this using `result.find_last_not_of('0')`.

File: src/util/ieee_float.cpp (hunk 1 above)
Comment (kroening) [inline src/util/ieee_float.cpp]: I don't actually know where that is from (was there before); and actual implementations appear to be different.

Comment (tautschnig) [inline src/util/ieee_float.cpp]: Would you mind adding a comment to that effect? In particular, the fact that actual implementations appear to differ sounds interesting to me.

**Pull request merged**